src/lib/datocms.ts

```diff
@@ -20,7 +20,7 @@ type DatocmsRequest = {
  */
 export const datocmsRequest = async <T>({ query, variables = {}, retryCount = 1 }: DatocmsRequest): Promise<T> => {
   const headers = new Headers({
-    Authorization: getSecret('DATOCMS_READONLY_API_TOKEN'),
+    Authorization: getSecret('DATOCMS_READONLY_API_TOKEN')!,
```

Comment:
`getSecret('DATOCMS_READONLY_API_TOKEN')` is possibly `undefined`, but I think it should always be set right? If it wouldn't be set then you wouldn't be able to get data.

Reply:
What happens when it is not set? Does it throw a nice error that the environment variable is missing, or do I only see the authorization fail? As a user I would be best served with an error that I'm missing the `.env`.

Reply:
Yeah I agree, but I also thought that one of the first things you do is adding a `.env` file. The error I get now doesn't really tell you what's wrong. Maybe we should put a warning in a sort of `predev` script?

```
luuk@Luuks-MBP head-start % npm run dev

> head-start@0.0.1 predev
> run-s prep


> head-start@0.0.1 prep
> run-s prep:* --print-label

[prep:clean-astro-env      ]
[prep:clean-astro-env      ] > head-start@0.0.1 prep:clean-astro-env
[prep:clean-astro-env      ] > rm -rf node_modules/.astro
[prep:clean-astro-env      ]
[prep:cloudflare-env       ]
[prep:cloudflare-env       ] > head-start@0.0.1 prep:cloudflare-env
[prep:cloudflare-env       ] > touch .dev.vars
[prep:cloudflare-env       ]
[prep:datocms-types        ]
[prep:datocms-types        ] > head-start@0.0.1 prep:datocms-types
[prep:datocms-types        ] > graphql-codegen --config .graphqlrc.ts
[prep:datocms-types        ]
[prep:datocms-types        ] (node:35547) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
[prep:datocms-types        ] (Use `node --trace-deprecation ...` to show where the warning was created)
[prep:datocms-types        ] Saving generated types for DatoCMS (environment: 'page-model-names') to 'src/lib/types/datocms.d.ts'.
[prep:datocms-types        ] [STARTED] Parse Configuration
[prep:datocms-types        ] [SUCCESS] Parse Configuration
[prep:datocms-types        ] [STARTED] Generate outputs
[prep:datocms-types        ] [STARTED] Generate to src/lib/types/datocms.d.ts
[prep:datocms-types        ] [STARTED] Load GraphQL schemas
[prep:datocms-types        ] [FAILED] Could not obtain introspection result, received the following as response;
[prep:datocms-types        ] [FAILED]  { data: [{ id: "5413bc", type: "api_error", attributes: [Object] }] }
[prep:datocms-types        ] [SUCCESS] Generate outputs
ERROR: "prep:datocms-types" exited with 1.
ERROR: "prep" exited with 1.
luuk@Luuks-MBP head-start %
```

Reply:
`dotenv-safe` already safeguards the `prep` scripts. Those should be able to function standalone.

For the `astro.config.ts` we could do the same, and throw a meaningful error in there to ensure all env variables are defined and that we don't need to check elsewhere in the code.